Tests/StreamChatTestTools/Mocks/StreamChat/ChatClientUpdater_Mock.swift

```diff
@@ -5,7 +5,7 @@
 @testable import StreamChat
 
 /// Mock implementation of `ChatClientUpdater`
-class ChatClientUpdaterMock: ChatClientUpdater {
+final class ChatClientUpdaterMock: ChatClientUpdater {
```

Comment:
@martinmitrevski that looks like a code-smell that we want to invest our time in. 
Mocks should never be subclassed - that's anti-pattern that comes with maintenance cost.

Can you tell me in which case you had to do subclass mock class?

Reply:
@bielikb my thinking was actually related to the UIKit SDK, not SwiftUI. We generally allow customizations there by subclassing and we can expect integrators to want mocks with the possibility to extend them based on their needs.

Reply:
@martinmitrevski subclassing mocks is not really the way to go. I feel like this is some premature optimization that just adds more maintenance on our side == not to forget to mark newly added vars/funcs in mocks as `open`.

In the provided way they're not completely blocked in achieving their customization - in such edge case they can still subclass from the main parent class.

I'll leave these marked as final for now. If we ever get through the case when we need to extend this further, we might revisit the way how we build/generate mocks.

Reply:
Yes, that fine with me. Let's see how are customers will be using them and if there's going to be demand for customising the mocks. But now that you mentioned it, generating mocks is something we should definitely check in the future, it will help us a lot in scenarios like this.

Reply:
There's a planned task for that in our epic. Hopefully we will get to that research already in Q2. lets keep our fingers crossed :)